### PR TITLE
add function serverInfo and serverInfoReset

### PR DESCRIPTION
**Pull request by Newcomer1989**
added missing functions analog to the channelList function

### CODE REVIEW

File: libraries/TeamSpeak3/Node/Server.php [hunk 1]
```diff
@@ -89,6 +89,28 @@ public function request($cmd, $throw = TRUE)
...
+  
+  /**
+   * Returns an array filled with TeamSpeak3_Node_Server objects.
+   *
+   * @param  array $filter
+   * @return array
+   */
+  public function serverInfo()
+  {
+	return $this->request("serverinfo")->toList();
```
Comment (ronindesign) [inline libraries/TeamSpeak3/Node/Server.php]: Looks like this is handled by: `TeamSpeak3_Node_Abstract->getInfo()`.
Ref: https://docs.planetteamspeak.com/ts3/php/framework/class_team_speak3___node___abstract.html#ac60d6af1acba9d78d719d4b86e987c07

Example to get server properties for 'virtual server' with id `1`:
```php
require_once 'TeamSpeak3.php';

$ts3_host = TeamSpeak3::factory("serverquery://username:password@255.255.255.255:10011");
$ts3_virtual_server = $ts3_host->serverGetById(1); // Where `1` is desired virtual server ID.

print_r($ts3_virtual_server->getInfo(true, true)); // Arguments specify extended, convert to human readable.
```

File: libraries/TeamSpeak3/Node/Server.php [hunk 2]
```diff
@@ -89,6 +89,28 @@ public function request($cmd, $throw = TRUE)
...
+  
+  /**
+   * Resets the server info.
+   *
+   * @return void
+   */
+  public function serverInfoReset()
+  {
+    $this->resetNodeList();
+    $this->serverInfo = null;
```
Comment (ronindesign) [inline libraries/TeamSpeak3/Node/Server.php]: `$this->serverInfo` is undefined in the context of `TeamSpeak3_Node_Server`.

Maybe you meant `$this->nodeInfo` defined in `TeamSpeak3_Node_Abstract`. In which case, you already have access to reset via `TeamSpeak3_Node_Server->resetNodeInfo()` as defined in: `TeamSpeak3_Node_Abstract->resetNodeInfo()`.
Ref: https://docs.planetteamspeak.com/ts3/php/framework/class_team_speak3___node___abstract.html#af0a532082869f2eb5c99951c4e9251b4

Also, note that `TeamSpeak3_Node_Server->nodeInfo` is initialized and reset to `(array) array()` not `null`.

**Comment by ronindesign** [comment]
@Newcomer1989 functionality already exists. See code comments.

**Comment by svenpaulsen** [comment]
Correct. This is already possible using the **getInfo()** methods.

**Comment by Newcomer1989** [comment]
Its very bad documentated on this position. How should I found this function, when I know the needed information are in serverquery command "serverinfo"?
Then I have a request. Please documentate this anywhere that getInfo() gives the "serverinfo" values back. Perhaps an easy table query command -> php framework function would help much, also for other people!

**Comment by ronindesign** [comment]
I like the idea of a table for reference!

Depending on what IDE / dev environment you work in, it's usually not too hard to follow the path of inheritance. For example, type-ahead autocompletion let's me type `$server->info` and then autocomplete shows all possible methods, including inherited ones, so I can do a quick search that way. Also, there are "structure" views in my IDE that show method and property inheritance as well.

Otherwise, when I'm looking for a method that might provide some needed Server Query command info, I typically go to the documentation page and start with the highest relevant node: `TeamSpeak3_Node_Abstract` If not there, then I check the first child in direction of my needed info's scope, so maybe `TeamSpeak3_Node_Client` would be next.

Sometimes, this doesn't work though and I definitely have just simply tried searching all of the source code for any reference to 'serverinfo'.

Maybe it would also be great to build out more examples, or groups of examples (e.g. some for working with clients, some for working with channels, etc).
